operator/src/main/java/oracle/kubernetes/operator/calls/AsyncRequestStep.java

```diff
@@ -377,14 +377,18 @@ public NextAction doPotentialRetry(Step conflictStep, Packet packet, int statusC
         }
 
         NextAction na = new NextAction();
-        if (statusCode == 0 && retryCount <= maxRetryCount) {
-          na.invoke(Optional.ofNullable(conflictStep).orElse(retryStep), packet);
-        } else {
-          LOGGER.finer(MessageKeys.ASYNC_RETRY, identityHash(), String.valueOf(waitTime));
-          na.delay(retryStep, packet, waitTime, TimeUnit.MILLISECONDS);
+        if (retryCount <= maxRetryCount) {
```

Comment:
We're now testing this condition tries, so it is worth breaking it out into its own method:```

    private boolean retriesLeft() {
      return retryCount <= maxRetryCount;
    };```

In addition, rather than using a nested if, negate the above condition and make it the first branch if an if-else if-else sequence, that simply returns na.

Reply:
I have made the suggested changes, please review, Thanks.